src/extension.ts

```diff
@@ -164,3 +177,34 @@ function convertLogLevel(commonLogLevel: string) {
             return "FINE";
     }
 }
+
+function resolvePotentialProjectFiles() {
+    let editors = vscode.window.visibleTextEditors.filter((e) => e.document.languageId === "java");
+    const activeEditor = vscode.window.activeTextEditor;
+    if (activeEditor && activeEditor.document.languageId === "java") {
+        const index = editors.indexOf(activeEditor);
+        editors = editors.splice(index, 1);
+        editors = [activeEditor].concat(editors);
+    }
+    if (editors) {
+        const projects = editors.map((e) => {
+            if (e.document.uri.scheme !== "file") {
+                return null;
+            }
+            const workspaceRoot = vscode.workspace.rootPath;
+            const fullpath = e.document.fileName;
+            let cur = path.dirname(fullpath);
+            while (!fs.existsSync(path.join(cur, ".project"))) {
+                if (path.resolve(cur) === workspaceRoot) {
+                    break;
+                }
+                cur = path.join(cur, "../");
+            }
+            if (fs.existsSync(path.join(cur, ".project"))) {
+                return path.resolve(cur, ".project");
+            }
+            return null;
+        });
+        return projects.filter((elem, pos) => elem !== null && projects.indexOf(elem) === pos);
```

Comment:
Why not to use `projects.filter((element) -> element != null)` directly?

Reply:
it also remove duplicate